taichi/program/sparse_matrix.h

```diff
@@ -211,19 +212,64 @@ class CuSparseMatrix : public SparseMatrix {
     }
 #endif
   }
+  explicit CuSparseMatrix(cusparseSpMatDescr_t A,
+                          int rows,
+                          int cols,
+                          DataType dt)
+      : SparseMatrix(rows, cols, dt), matrix_(A) {
+  }
+  CuSparseMatrix(const CuSparseMatrix &sm)
+      : SparseMatrix(sm.rows_, sm.cols_, sm.dtype_), matrix_(sm.matrix_) {
+  }
 
   virtual ~CuSparseMatrix();
+
+  // TODO: Overload +=, -= and *=
+  friend std::unique_ptr<SparseMatrix> operator+(const CuSparseMatrix &lhs,
+                                                 const CuSparseMatrix &rhs) {
+    auto m = lhs.addition(rhs, 1.0, 1.0);
+    return m;
+  };
+
+  friend std::unique_ptr<SparseMatrix> operator-(const CuSparseMatrix &lhs,
+                                                 const CuSparseMatrix &rhs) {
+    return lhs.addition(rhs, 1.0, -1.0);
+  };
+
+  friend std::unique_ptr<SparseMatrix> operator*(const CuSparseMatrix &sm,
+                                                 float scale) {
+    return sm.addition(sm, scale, 0.0);
+  }
+
+  friend std::unique_ptr<SparseMatrix> operator*(float scale,
+                                                 const CuSparseMatrix &sm) {
+    return sm.addition(sm, scale, 0.0);
+  }
+
+  std::unique_ptr<SparseMatrix> addition(const CuSparseMatrix &other,
+                                         const float alpha,
```

Comment:
nit: is `const float` intended?

Reply:
The only use of `const` here, I think, is to prevent us from modifying the `alpha` or `beta` during implement this function. :-)